File: leaderboard-js/src/Leaderboard.js
var gameInfo = function() {
  return [
   {
     home_team: "Patriots",
     away_team: "Broncos",
     home_score: 7,
     away_score: 3
   },
   {
     home_team: "Broncos",
     away_team: "Colts",
     home_score: 3,
     away_score: 0
   },
   {
     home_team: "Patriots",
     away_team: "Colts",
     home_score: 11,
     away_score: 7
   },
   {
     home_team: "Steelers",
     away_team: "Patriots",
     home_score: 7,
     away_score: 21
   }
 ]
}

// YOUR CODE HERE
function TeamInfo(name, rank, wins, losses) {
  this.name = name;
  this.rank = rank;
  this.wins = wins;
  this.losses = losses;
}


var team_names = []
var winners = []
var lossers = []
for(var i = 0; i < gameInfo().length; i++){
  team_names.push(gameInfo()[i].home_team)
  team_names.push(gameInfo()[i].away_team)
  if (gameInfo()[i].home_score > gameInfo()[i].away_score) {
    winners.push(gameInfo()[i].home_team)
    lossers.push(gameInfo()[i].away_team)
  }
  else {
    lossers.push(gameInfo()[i].home_team)
    winners.push(gameInfo()[i].away_team)
  }
}
var uniq_names = Array.from(new Set(team_names));
var team_info = []
for(var i = 0; i < uniq_names.length; i++){
  team_info.push(new TeamInfo(uniq_names[i], 0, 0, 0))
  }


for(var j = 0; j < team_info.length; j++){
  for(var i = 0; i < winners.length; i++){
      if (winners[i] === team_info[j].name){
        team_info[j].wins = team_info[j].wins + 1;
      }
      if (lossers[i] === team_info[j].name){
        team_info[j].losses = team_info[j].losses + 1;
      }
  }
}

team_info.sort(function(a, b){
    return b.wins-a.wins
})
team_info.sort(function(a, b){
    return a.losses-b.losses
})

for(var i = 0; i < team_info.length; i++){
  team_info[i].rank = i + 1;
}

console.log("Teams\n " + Array(34).join("-"));
console.log("| Rank " + "  Name" + "        Wins" + "   Losses|");
for(var i = 0; i < team_info.length; i++){
  var empty_space = 15 - team_info[i].name.length;
  console.log("| " + team_info[i].rank + ".   " + team_info[i].name +
  Array(empty_space).join(" ") + team_info[i].wins +
   "      " + team_info[i].losses + "     " + "|");
}
console.log(" " +Array(34).join("-"));


var gameSum = ""
function teamSummary(name){
  for(var i = 0; i < team_info.length; i++){
      if (team_info[i].name === name) {
        var teamSum = (team_info[i].name + " - Summary:\n" + "Rank: " +
        team_info[i].rank + "\nWins: " + team_info[i].wins +
      "\nLosses: " + team_info[i].losses);
      }
  }
  for(var j = 0; j < gameInfo().length; j++){
    if (gameInfo()[j].home_team === name || gameInfo()[j].away_team === name){
      gameSum = gameSum + (gameInfo()[j].home_team + "  " +  gameInfo()[j].home_score +
      " : " + gameInfo()[j].away_score + "  " + gameInfo()[j].away_team + "\n");
    }
  }
  return (teamSum + "\nGames: \n" + gameSum)
}

console.log(teamSummary("Patriots"));

File: leaderboard/lib/leaderboard.rb
require_relative "../spec/spec_helper"

class Leaderboard
GAME_INFO = [
    {
      home_team: "Patriots",
      away_team: "Broncos",
      home_score: 17,
      away_score: 13
    },
    {
      home_team: "Broncos",
      away_team: "Colts",
      home_score: 24,
      away_score: 7
    },
    {
      home_team: "Patriots",
      away_team: "Colts",
      home_score: 21,
      away_score: 17
    },
    {
      home_team: "Broncos",
      away_team: "Steelers",
      home_score: 11,
      away_score: 27
    },
    {
      home_team: "Steelers",
      away_team: "Patriots",
      home_score: 24,
      away_score: 31
    }
]

# YOUR CODE GOES HERE
  attr_reader :games,:teams, :winners
  #  attr_reader :teams
    def initialize
      @games = []
      @teams = []
      @winners = []
      @losers = []
      GAME_INFO.each do |game|
        @games << game
      end
    end


    def teams
      @games.each do |game|
        @teams << Team.new(game[:home_team])
        @teams << Team.new(game[:away_team])
      end
      @teams.uniq! do |team|
        team.name
      end
      @teams
    end

    def winners
      @games.each do |game|
        if game[:home_score] > game[:away_score]
          @winners << game[:home_team]
        else
          @winners << game[:away_team]
        end
      end
      @winners
    end

    def losers
      @games.each do |game|
        if game[:home_score] < game[:away_score]
          @losers << game[:home_team]
        else
          @losers << game[:away_team]
        end
      end
      @losers
    end

    def add_result
      @teams.each do |team|
        @winners.each do |winner|
          if winner == team.name
            team.wins += 1
          end
        end
        @losers.each do |loser|
          if loser == team.name
            team.losses += 1
          end
        end
      end
      @teams
    end

    def ranking
      @teams.sort_by {|team| team.wins}.reverse!
      @teams.sort_by! {|team| team.losses}
      @teams.each_with_index do |team, index|
        team.rank = index + 1
      end
    end

    def summary
      result =  " ---------------------------------------------------\n"
      result += "|  Name       Rank      Total Wins    Total Losses |\n"
      result += "|  #{(@teams[0]).name}   #{@teams[0].rank}          #{@teams[0].wins}             #{@teams[0].losses}           |\n"
      result += "|  #{(@teams[1]).name}   #{@teams[1].rank}          #{@teams[1].wins}             #{@teams[1].losses}           |\n"
      result += "|  #{(@teams[2]).name}    #{@teams[2].rank}          #{@teams[2].wins}             #{@teams[2].losses}           |\n"
      result += "|  #{(@teams[3]).name}      #{@teams[3].rank}          #{@teams[3].wins}             #{@teams[3].losses}           |\n"
      result +=  " ---------------------------------------------------\n"

    end
end
